tools/lemmacmd/lemmacmd.go

```diff
@@ -0,0 +1,211 @@
+	in := fs.String("in", "", "path to file to be read in")
+	out := fs.String("out", "", "path to file to be written out")
+	key := fs.String("keypath", "", "path to base64-encoded 32-byte key on disk, if no path is provided, a passphrase will be used")
+	iter := fs.Int("itercount", 524288, "if a passphrase is used, iteration count for pbkdf#2")
+
+	err := fs.Parse(args[2:])
+	if err != nil {
+		fmt.Printf("lemmacmd: unable to parse flags: %v\n", err)
+	}
+
+	if mode != "encrypt" && mode != "decrypt" {
+		fmt.Printf("lemmacmd: mode must be encrypt or decrypt, not: %q\n", mode)
+	}
+	if *in == "" {
+		fmt.Printf("lemmacmd: input path required\n")
+	}
+	if *out == "" {
+		fmt.Printf("lemmacmd: output path required\n")
+	}
+	if *in == "" || *out == "" {
+		usage()
+		os.Exit(255)
+	}
+
+	return mode, *key, *iter, *in, *out
+}
+
+func encrypt(keypath string, itercount int, inputpath string, outputpath string) {
+	salt, err := (&random.CSPRNG{}).Bytes(16)
+	if err != nil {
+		fmt.Printf("lemmacmd: unable to generate salt: %v\n", err)
+		os.Exit(255)
+	}
+
+	key, err := generateKey(keypath, salt, itercount)
+	if err != nil {
+		fmt.Printf("lemmacmd: unable to generate or read in key: %v\n", err)
+		os.Exit(255)
+	}
+
+	plaintextBytes, err := ioutil.ReadFile(inputpath)
+	if err != nil {
+		fmt.Printf("lemmacmd: unable to read plaintext file %q: %v\n", inputpath, err)
+		os.Exit(255)
+	}
+
+	sealedData, err := secret.Seal(plaintextBytes, key)
+	if err != nil {
+		fmt.Printf("lemmacmd: unable to seal plaintext: %v\n", err)
+		os.Exit(255)
+	}
+
+	err = writeCiphertext(salt, itercount, sealedData, outputpath)
+	if err != nil {
+		fmt.Printf("lemmacmd: unable to write sealed data to disk: %v\n", err)
+		os.Exit(255)
+	}
+}
+
+func decrypt(keypath string, inputpath string, outputpath string) {
+	keySalt, keyIter, sealedData, err := readCiphertext(inputpath)
+	if err != nil {
+		fmt.Printf("lemmacmd: unable to read ciphertext file: %v\n", err)
+		os.Exit(255)
+	}
+
+	key, err := generateKey(keypath, keySalt, keyIter)
+	if err != nil {
+		fmt.Printf("lemmacmd: unable to build key: %v\n", err)
+		os.Exit(255)
+	}
+
+	plaintextBytes, err := secret.Open(sealedData, key)
+	if err != nil {
+		fmt.Printf("lemmacmd: unable to open ciphertext: %v\n", err)
+		os.Exit(255)
+	}
+
+	err = ioutil.WriteFile(outputpath, plaintextBytes, 0600)
+	if err != nil {
+		fmt.Printf("lemmacmd: unable to write plaintext bytes to disk: %v\n", err)
+		os.Exit(255)
+	}
+}
+
+// Returns key from keypath or uses salt + passphrase to generate key using a KDF.
+func generateKey(keypath string, salt []byte, keyiter int) (key *[secret.SecretKeyLength]byte, err error) {
+	// if a keypath is given try and use it
+	if keypath != "" {
+		key, err := secret.ReadKeyFromDisk(keypath)
+		if err != nil {
+			return nil, fmt.Errorf("unable to build secret service: %v", err)
+		}
+		return key, nil
+	}
+
+	// otherwise read in a passphrase from disk and use that, remember to reset your terminal afterwards
```

Comment:
You could use https://github.com/howeyc/gopass to avoid echoing in the first place